core/modal_models/modes.h

```diff
@@ -90,18 +109,11 @@ void _lf_handle_mode_changes(void);
  */
 void _lf_handle_mode_triggered_reactions(void);
 
-////////////////////////////////////////////////////////////
-//// Type definitions for modal infrastructure.
-
-/** Typedef for reactor_mode_t struct, used for representing a mode. */
-typedef struct reactor_mode_t reactor_mode_t;
-/** Typedef for reactor_mode_state_t struct, used for storing modal state of reactor and/or its relation to enclosing modes. */
-typedef struct reactor_mode_state_t reactor_mode_state_t;
-/** Typedef for mode_state_variable_reset_data_t struct, used for storing data for resetting state variables nested in modes. */
-typedef struct mode_state_variable_reset_data_t mode_state_variable_reset_data_t;
+/**
+ * Function (to be code generated) to fill up a vector with the reactors which have mode transitions.
```

Comment:
More information is needed here.  From the usage of this function, I infer that a new vector will be allocated (which never gets freed?) and that it only adds reactors which have pending mode transitions, not all reactors with modes. Is this correct? Please clarify and explain how the argument should be used (if a new vector is allocated, it must later be freed).

Reply:
Have clarified. Vector is allocated and freed inside `scheduler_LET.c`.